eventually-core/src/aggregate.rs

```diff
@@ -150,7 +150,7 @@ where
     T: Aggregate + 'static,
 {
     id: T::Id,
-    version: u32,
+    version: i64,
```

Comment:
this seems slightly odd to me. *Semantically* a signed integer doesn't make any sense here, and *functionally* `u32` and `i64` hold the same number of (positive) values. Agree that avoiding the panics is useful, but this doesn't feel to me like the right way to do it.

but i'm open to being convinced

Reply:
There are two reasons why I chose i64:

- All the database connectors I know cast database big integers to i64, as they can be indeed negative. I chose `i64` to bring struct types closer to what database clients will give
- By mapping it to i64 each EventStore has total freedom to implement this in the way they want. I'll give an example. The default value in Rust for i64 is zero, which means that the default version would be `0` too. However, that may be confused with the first version of the event if a zero-index is used. If I ask myself: what is the version of an aggregate with no events? I may answer `-1`